rmw_opendds_cpp/src/DDSSubscriber.cpp

```diff
@@ -122,10 +121,10 @@ void DDSSubscriber::cleanup()
 }
 
 DDSSubscriber::DDSSubscriber(DDS::DomainParticipant_var dp
-  , const rosidl_message_type_support_t & ros_ts
+  , const rosidl_message_type_support_t * ros_ts
   , const char * topic_name
-  , const rmw_qos_profile_t & rmw_qos
-) : DDSEntity(ros_ts, topic_name, rmw_qos)
+  , const rmw_qos_profile_t * rmw_qos
```

Comment:
Just curious why we switched from by ref to by pointer.

Reply:
to localize the null-checking and the following creation code in DDSTopic for safety and consistency for both pub and sub.